packages/hoppscotch-app/helpers/import-export/import/openapi/exampleV2.ts

```diff
@@ -0,0 +1,188 @@
+import { OpenAPIV2 } from "openapi-types"
+import * as O from "fp-ts/Option"
+import { pipe, flow } from "fp-ts/function"
+import * as A from "fp-ts/Array"
+import { prettyPrintJSON } from "~/helpers/functional/json"
+
+type PrimitiveSchemaType = "string" | "integer" | "number" | "boolean"
+
+type SchemaType = "array" | "object" | PrimitiveSchemaType
+
+type PrimitiveRequestBodyExampleType = number | string | boolean
+
+type RequestBodyExampleType =
+  | { [name: string]: RequestBodyExampleType }
+  | Array<RequestBodyExampleType>
+  | PrimitiveRequestBodyExampleType
+
+const getPrimitiveTypePlaceholder = (
+  schemaType: PrimitiveSchemaType
+): PrimitiveRequestBodyExampleType => {
+  switch (schemaType) {
+    case "string":
+      return "string"
+    case "integer":
+    case "number":
+      return 1
+    case "boolean":
+      return true
+  }
+}
+
+const getSchemaTypeFromSchemaObject = (
+  schema: OpenAPIV2.SchemaObject
+): O.Option<SchemaType> =>
+  pipe(
+    schema.type,
+    O.fromNullable,
+    O.map(
+      (schemaType) =>
+        (Array.isArray(schemaType) ? schemaType[0] : schemaType) as SchemaType
+    )
+  )
+
+const isSchemaTypePrimitive = (
+  schemaType: string
+): schemaType is PrimitiveSchemaType =>
+  ["string", "integer", "number", "boolean"].includes(schemaType)
+
+const isSchemaTypeArray = (schemaType: string): schemaType is "array" =>
+  schemaType === "array"
+
+const isSchemaTypeObject = (schemaType: string): schemaType is "object" =>
+  schemaType === "object"
```

Comment:
Can you explain the type guard comment in the last few lines? If `isXXX` were to accept schema object, then the output type would be boolean.

Reply:
I am meaning, we can write a single function with this signature:
```ts
type SchemaType = "primitive" | "object" | "array"
const isSchemaOfType = (type: SchemaType) => (schema: OpenAPIV2.SchemaObject): boolean => { // ... }
```
Because, in the occurances of both `getSchemaTypeFromSchemaObject` and `isSchemaTypeXXX` always happen together (except in one place alone https://github.com/hoppscotch/hoppscotch/pull/2243/files#diff-086cd60388447b0235856b196f4cb3bebf774807763bd282bc210cb61d061d4cR85). Also, we are not utilizing the type guard as well, so we can just roll it down to a function that returns a boolean instead.

With this refactor many places become more clean. For example,
Before:
```ts
const arrayTypeExample = pipe(
    schema,
    O.fromPredicate(
      flow(
        getSchemaTypeFromSchemaObject,
        O.map(isSchemaTypeArray),
        O.getOrElseW(() => false) // No schema type found in the schema object, assume type to be different from array
      )
    ),
    O.map((schema) => schema.items as OpenAPIV2.ItemsObject),
    O.map(generateExampleArrayFromOpenAPIV2ItemsObject)
  )
```

After:
```ts
const arrayTypeExample = pipe(
    schema,
    O.fromPredicate(isSchemaOfType("array")),
    O.map((schema) => schema.items as OpenAPIV2.ItemsObject),
    O.map(generateExampleArrayFromOpenAPIV2ItemsObject)
  )
```

Reply:
Gotcha! Will make this refactor and push the changes.